src/Project.ts

```diff
@@ -1,11 +1,30 @@
-import Trigger from "./Trigger.js";
-import Renderer from "./Renderer.js";
-import Input from "./Input.js";
-import LoudnessHandler from "./Loudness.js";
-import Sound from "./Sound.js";
+import Trigger, { TriggerOptions } from "./Trigger";
+import Renderer from "./Renderer";
+import Input from "./Input";
+import LoudnessHandler from "./Loudness";
+import Sound from "./Sound";
+import type { Stage, Sprite } from "./Sprite";
+
+type TriggerWithTarget = {
+  target: Sprite | Stage;
+  trigger: Trigger;
+};
 
 export default class Project {
-  constructor(stage, sprites = {}, { frameRate = 30 } = {}) {
+  public stage: Stage;
+  public sprites: Partial<Record<string, Sprite>>;
```

Comment:
`Record<string, T>` assumes that *every* string key maps to a valid `T`. When we access a sprite from `Project.sprites` by name, that sprite may not exist.

Unlike `sb-edit`, I enabled TypeScript's strict mode here, so it's stricter about checking for null and undefined values, and I wanted to make use of that by making sure that we don't assume object keys always map to a value.

It might be better to have `Project` take a type parameter listing the valid sprite names, so that we can statically know whether or not a sprite exists and error only if the user tries to access a nonexistent sprite.

Reply:
Got it, that makes sense. Do you feel it'd be worth enabling strict mode in sb-edit (once we're done with this PR) and being more conscious of undefined/null there?

That static erroring sounds practical from a user's perspective. I haven't dug into the code for targets yet, but is it possible to get the same effect for variables (i.e. error when accessing nonexistant variables)? Maybe that's something we're already doing and/or free.

Reply:
Yes; either after or as part of the sb-edit TypeScript upgrade, I plan on enabling strict mode and fixing any resulting errors there.

I've already done something similar for sprites' variables here. I ran into an issue when trying to do it for `Project`, though--if I make the project's sprites a type parameter, the `Project` constructor can't have a default value for the sprites argument ([it's complicated](https://stackoverflow.com/questions/56505560/how-to-fix-ts2322-could-be-instantiated-with-a-different-subtype-of-constraint)).

Reply:
Got it, thanks for the reference info!